.github/workflows/publish_plugin_to_hub_snowflake.yml

```diff
@@ -0,0 +1,115 @@
+name: Publish plugin to Snowflake to hub
+on:
+  push:
+    tags:
+      - "plugins-destination-snowflake-v*.*.*"
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      plugin_name: ${{ steps.split.outputs.plugin_name }}
+      plugin_kind: ${{ steps.split.outputs.plugin_kind }}
+      plugin_version: ${{ steps.split.outputs.plugin_version }}
+      plugin_dir: ${{ steps.split.outputs.plugin_dir }}
+      prerelease: ${{ steps.semver_parser.outputs.prerelease }}
+      plugin_releaser: ${{ steps.set-result.outputs.result }}
+    steps:
+      - name: Split tag
+        id: split
+        run: |
+          tag=${{ github.ref_name }}
+          plugin_kind=$(echo $tag | cut -d- -f2)
+          plugin_name=$(echo $tag | cut -d- -f3)
+          plugin_version=$(echo $tag | cut -d- -f4-)
+          echo "plugin_name=${plugin_name}" >> $GITHUB_OUTPUT
+          echo "plugin_kind=${plugin_kind}" >> $GITHUB_OUTPUT
+          echo "plugin_version=${plugin_version}" >> $GITHUB_OUTPUT
+          echo "plugin_dir=plugins/${plugin_kind}/${plugin_name}" >> $GITHUB_OUTPUT
+      # Fail if not a valid SemVer string
+      - name: Parse semver string
+        uses: booxmedialtd/ws-action-parse-semver@7784200024d6b3fc01253e617ec0168daf603de3
+        id: semver_parser
+        with:
+          input_string: ${{steps.split.outputs.plugin_version}}
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/github-script@v6
+        id: set-result
+        env:
+          PLUGIN_DIR: ${{steps.split.outputs.plugin_dir}}
+        with:
+          script: |
+            const fs = require('fs').promises;
+            const path = require('path');
+            const pluginFiles = await fs.readdir(process.env.PLUGIN_DIR);
+            if (pluginFiles.includes('Dockerfile')) {
+              return 'docker';
+            }
+            if (pluginFiles.includes('.goreleaser.yaml')) {
+              return 'go';
+            }
+          result-encoding: string
+  publish-plugin-to-hub:
+    timeout-minutes: 60
+    runs-on: large-ubuntu-monorepo
+    container:
+      image: ghcr.io/cloudquery/golang-cross:v10.0.0
+      env:
```

Comment:
A mix of https://github.com/cloudquery/cloudquery/blob/54fc73e3b0993ee132d0750a1b24084d2c775638/plugins/destination/snowflake/.goreleaser.yaml#L20 and https://github.com/cloudquery/cloudquery/blob/54fc73e3b0993ee132d0750a1b24084d2c775638/plugins/destination/snowflake/.goreleaser.yaml#L43

Not sure if this works. If it doesn't we'll need to support different configurations per target